File: src/main/java/com/clearcont/clearcontapp/model/Controle.java
package com.clearcont.clearcontapp.model;

import com.clearcont.clearcontapp.helpers.DecimalFormatBR;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.jetbrains.annotations.NotNull;

import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Entity
@Getter
@Setter
public class Controle {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String subGrupo;
    private String circulante;
    private String nomeConta;
    private Double saldoBalancete;
    private Double saldoAnalise;
    private Double valorDiferenca;
    private String statusGeralConciliacao;
    private String observacoes;
    private Boolean composicaoPreenchida;
    private Boolean agingListadaPendencia;
    private LocalDate dataCompetencia;
    private String nomeResponsavel;
    @ManyToOne
    private Empresa empresa;
    
    public Double getDoubleSaldoBalancete() {
        return saldoBalancete;
    }
    
    public Double getDoubleSaldoAnalise() {
        return saldoAnalise;
    }
    
    public Double getDoubleValorDiferenca() {
        return valorDiferenca;
    }
    
    public @NotNull String getSaldoBalancete() {
        return DecimalFormatBR.getDecimalFormat().format(saldoBalancete);
    }
    
    public @NotNull String getSaldoAnalise() {
        return DecimalFormatBR.getDecimalFormat().format(saldoAnalise);
    }
    
    public @NotNull String getValorDiferenca() {
        return DecimalFormatBR.getDecimalFormat().format(valorDiferenca);
    }
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/models/Controle.kt
package br.com.clearcont.clearcontwebapp.models

import br.com.clearcont.clearcontwebapp.helpers.formatCurrencyBR
import br.com.clearcont.clearcontwebapp.models.enums.TypeCount
import jakarta.persistence.*
import java.time.LocalDate

@Entity
class Controle(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    val subGrupo: String? = null,
    val circulante: TypeCount? = null,
    @Column(length = 30)
    val nomeConta: String? = null,
    val doubleSaldoBalancete: Double = 0.0,
    val doubleSaldoAnalise: Double= 0.0,
    val doubleValorDiferenca: Double= 0.0,
    val statusGeralConciliacao: String? = null,
    val observacoes: String? = null,
    val composicaoPreenchida: Boolean? = null,
    val agingListadaPendencia: Boolean? = null,
    val dataCompetencia: LocalDate? = null,
    val nomeResponsavel: String? = null,
    @ManyToOne
    val empresa: Empresa? = null
) {

    fun getSaldoBalancete(): String {
        return formatCurrencyBR(doubleSaldoBalancete)
    }

    fun getSaldoAnalise(): String {
        return formatCurrencyBR(doubleSaldoAnalise)
    }

    fun getValorDiferenca(): String {
        return formatCurrencyBR(doubleValorDiferenca)
    }
}
